src/reflect/value.go

```diff
@@ -355,13 +352,16 @@ func (v Value) Len() int {
 	case Array:
 		return v.typecode.Len()
 	case Chan:
-		return chanlen(v.value)
+		return chanlen(v.pointer())
 	case Map:
-		return maplen(v.value)
+		return maplen(v.pointer())
 	case Slice:
-		return int((*sliceHeader)(v.value).len)
+		if v.pointer() == nil {
+			return 0
```

Comment:
A slice is always stored as a pointer in reflect.Value (because it's bigger than a pointer). So I think `v.value == nil` would be more appropriate.
The same is true below: I think `int((*sliceHeader)(v.value).len)` is safe.